File: app/src/main/java/com/mappls/sdk/demo/java/activity/DirectionWidgetActivity.java
package com.mappls.sdk.demo.java.activity;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.mappls.sdk.demo.R;
import com.mappls.sdk.demo.databinding.ActivityDirectionWidgetBinding;
import com.mappls.sdk.demo.java.settings.MapplsDirectionWidgetSetting;
import com.mappls.sdk.direction.ui.DirectionCallback;
import com.mappls.sdk.direction.ui.DirectionFragment;
import com.mappls.sdk.direction.ui.model.DirectionOptions;
import com.mappls.sdk.direction.ui.model.DirectionPoint;
import com.mappls.sdk.plugins.places.autocomplete.model.PlaceOptions;
import com.mappls.sdk.services.api.directions.models.DirectionsResponse;

import java.util.List;

public class DirectionWidgetActivity extends AppCompatActivity {

    private ActivityDirectionWidgetBinding mBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_direction_widget);

        DirectionOptions.Builder optionsBuilder = DirectionOptions.builder();
        if (!MapplsDirectionWidgetSetting.getInstance().isDefault()){

            PlaceOptions options = PlaceOptions.builder()
                    .zoom(MapplsDirectionWidgetSetting.getInstance().getZoom())
                    .hint(MapplsDirectionWidgetSetting.getInstance().getHint())
                    .location(MapplsDirectionWidgetSetting.getInstance().getLocation())
                    .filter(MapplsDirectionWidgetSetting.getInstance().getFilter())
                    .saveHistory(MapplsDirectionWidgetSetting.getInstance().isEnableHistory())
                    .pod(MapplsDirectionWidgetSetting.getInstance().getPod())
                    .attributionHorizontalAlignment(MapplsDirectionWidgetSetting.getInstance().getSignatureVertical())
                    .attributionVerticalAlignment(MapplsDirectionWidgetSetting.getInstance().getSignatureHorizontal())
                    .logoSize(MapplsDirectionWidgetSetting.getInstance().getLogoSize())
                    .tokenizeAddress(MapplsDirectionWidgetSetting.getInstance().isTokenizeAddress())
                    .historyCount(MapplsDirectionWidgetSetting.getInstance().getHistoryCount())
                    .backgroundColor(getResources().getColor(MapplsDirectionWidgetSetting.getInstance().getBackgroundColor()))
                    .toolbarColor(getResources().getColor(MapplsDirectionWidgetSetting.getInstance().getToolbarColor()))
                    .build();


            optionsBuilder.searchPlaceOption(options)
                    .showDefaultMap(true)
                    .showStartNavigation(MapplsDirectionWidgetSetting.getInstance().isShowStartNavigation())
                    .steps(MapplsDirectionWidgetSetting.getInstance().isSteps())
                    .resource(MapplsDirectionWidgetSetting.getInstance().getResource())
                    .profile(MapplsDirectionWidgetSetting.getInstance().getProfile())
                    .excludes(MapplsDirectionWidgetSetting.getInstance().getExcludes())
                    .overview(MapplsDirectionWidgetSetting.getInstance().getOverview())
            .showAlternative(MapplsDirectionWidgetSetting.getInstance().isShowAlternative())
            .searchAlongRoute(MapplsDirectionWidgetSetting.getInstance().isShowPOISearch());
        }

        DirectionFragment directionFragment = DirectionFragment.newInstance(optionsBuilder.build());



        getSupportFragmentManager().
                beginTransaction().
                add(mBinding.fragmentConatiner.getId(), directionFragment, DirectionFragment.class.getSimpleName()).
                commit();

        directionFragment.setDirectionCallback(new DirectionCallback() {
            @Override
            public void onCancel() {
                finish();
            }

            @Override
            public void onStartNavigation(DirectionPoint directionPoint, DirectionPoint directionPoint1, List<DirectionPoint> list, DirectionsResponse directionsResponse, int i) {
                Toast.makeText(DirectionWidgetActivity.this, "On Navigation Start", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/com/mappls/sdk/demo/kotlin/activity/DirectionWidgetActivity.kt
package com.mappls.sdk.demo.kotlin.activity

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.mappls.sdk.demo.R
import com.mappls.sdk.demo.databinding.ActivityDirectionWidgetBinding
import com.mappls.sdk.demo.kotlin.settings.MapplsDirectionWidgetSetting
import com.mappls.sdk.direction.ui.DirectionCallback
import com.mappls.sdk.direction.ui.DirectionFragment
import com.mappls.sdk.direction.ui.model.DirectionOptions
import com.mappls.sdk.direction.ui.model.DirectionPoint
import com.mappls.sdk.plugins.places.autocomplete.model.PlaceOptions
import com.mappls.sdk.services.api.directions.models.DirectionsResponse

class DirectionWidgetActivity : AppCompatActivity() {

    private lateinit var mBinding: ActivityDirectionWidgetBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_direction_widget)
        val optionsBuilder = DirectionOptions.builder()
        if (!MapplsDirectionWidgetSetting.instance.isDefault) {
            val options = PlaceOptions.builder()
                .zoom(MapplsDirectionWidgetSetting.instance.zoom)
                .hint(MapplsDirectionWidgetSetting.instance.hint)
                .location(MapplsDirectionWidgetSetting.instance.location)
                .filter(MapplsDirectionWidgetSetting.instance.filter)
                .saveHistory(MapplsDirectionWidgetSetting.instance.isEnableHistory)
                .pod(MapplsDirectionWidgetSetting.instance.pod)
                .attributionHorizontalAlignment(MapplsDirectionWidgetSetting.instance.signatureVertical)
                .attributionVerticalAlignment(MapplsDirectionWidgetSetting.instance.signatureHorizontal)
                .logoSize(MapplsDirectionWidgetSetting.instance.logoSize)
                .tokenizeAddress(MapplsDirectionWidgetSetting.instance.isTokenizeAddress)
                .historyCount(MapplsDirectionWidgetSetting.instance.historyCount)
                .backgroundColor(resources.getColor(MapplsDirectionWidgetSetting.instance.backgroundColor))
                .toolbarColor(resources.getColor(MapplsDirectionWidgetSetting.instance.toolbarColor))
                .build()
            optionsBuilder.searchPlaceOption(options)
                .showStartNavigation(MapplsDirectionWidgetSetting.instance.isShowStartNavigation)
                .steps(MapplsDirectionWidgetSetting.instance.isSteps)
                .resource(MapplsDirectionWidgetSetting.instance.resource)
                .profile(MapplsDirectionWidgetSetting.instance.profile)
                .excludes(MapplsDirectionWidgetSetting.instance.excludes)
                .overview(MapplsDirectionWidgetSetting.instance.overview)
                .showAlternative(MapplsDirectionWidgetSetting.instance.isShowAlternative)
                .searchAlongRoute(MapplsDirectionWidgetSetting.instance.isShowPOISearch)
        }

        val directionFragment = DirectionFragment.newInstance(optionsBuilder.build())
        supportFragmentManager.beginTransaction().add(
            mBinding.fragmentConatiner.id,
            directionFragment,
            DirectionFragment::class.java.simpleName
        ).commit()

        directionFragment.setDirectionCallback(object : DirectionCallback {
            override fun onCancel() {
                finish()
            }

            override fun onStartNavigation(
                directionPoint: DirectionPoint,
                directionPoint1: DirectionPoint,
                list: List<DirectionPoint>,
                directionsResponse: DirectionsResponse,
                i: Int
            ) {
                Toast.makeText(
                    this@DirectionWidgetActivity,
                    "On Navigation Start",
                    Toast.LENGTH_SHORT
                ).show()
            }
        })
    }
}